app/code/Magento/Theme/Controller/Result/MessagePlugin.php

```diff
@@ -171,11 +180,36 @@ private function setCookie(array $messages)
             }
 
             $publicCookieMetadata = $this->cookieMetadataFactory->createPublicCookieMetadata();
-            $publicCookieMetadata->setDurationOneYear();
-            $publicCookieMetadata->setPath($this->sessionConfig->getCookiePath());
+
+            if( $configLifetime = $this->scopeConfig->getValue(
+                'web/cookie/cookie_lifetime',
+                \Magento\Store\Model\ScopeInterface::SCOPE_STORE
+            ) ) {
+                $publicCookieMetadata->setDuration($configLifetime);
+            } else {
+                $publicCookieMetadata->setDurationOneYear();
+            }
+
+            if( $configPath = $this->scopeConfig->getValue(
+                'web/cookie/cookie_path',
+                \Magento\Store\Model\ScopeInterface::SCOPE_STORE
+            ) ) {
+                $publicCookieMetadata->setPath($configPath);
+            } else {
+                 $publicCookieMetadata->setPath($this->sessionConfig->getCookiePath());
+            }
+            
```

Comment:
I don't think this change is related to the issue you're trying to fix. Could you please revert it?
```suggestion
```